### ISSUE TITLE
Can Anyone run success this project?

### ISSUE THREAD

**Issue by John1231983** [AlexanderParkin/ChaLearn_liveness_challenge#9]
I think the author has not double check the repo. Many error and missing file in this repo. If someone has  sucess run  it, please let me know

**Comment by xudou3** [comment]
yeah, it will need to comment some parts for missing file and i changed some setting in opt file so it can run on my single 1070

**Comment by mrkstt** [comment]
Hi @John1231983 
I also successfully run the code. Probably this note can be useful:

1. Remark all mobilenetv2 import (there are three files, find using "find in file" option on visual code)

2. Make sure all paths are correct, including "data_root" at init_dataloader.py, weights paths at resnet_caffe_*.py, and etc.

3. Remark this part on "trainer.py":
""" self.visdom_log_file = os.path.join(self.opt.out_path, 'log_files', 'visdom.log')
        self.vis = Visdom(port = opt.visdom_port,
                          log_to_filename=self.visdom_log_file,
                          env=opt.exp_name + '_' + str(opt.fold))

        self.vis_loss_opts = {'xlabel': 'epoch', 
                              'ylabel': 'loss', 
                              'title':'losses', 
                              'legend': ['train_loss', 'val_loss']}

        self.vis_tpr_opts = {'xlabel': 'epoch', 
                              'ylabel': 'tpr', 
                              'title':'val_tpr', 
                              'legend': ['tpr@fpr10-2', 'tpr@fpr10-3', 'tpr@fpr10-4']}

        self.vis_epochloss_opts = {'xlabel': 'epoch', 
                              'ylabel': 'loss', 
                              'title':'epoch_losses', 
                              'legend': ['train_loss', 'val_loss']} """

4. Don't forget to download the model*.pth from CASIA-SURF dataset, and put on the corresponding folder as shown on point 5 (pth path).

5. I run the val script as follows:

python inference.py --config /home/face-spoofing/ChaLearn_liveness_challenge/data/opts/exp1_2stage_seed1/exp1_2stage_seed1_fold1_stage1.opt --pth /home/face-spoofing/ChaLearn_liveness_challenge/data/opts/exp1_2stage_seed1/fold1/checkpoints/model_30.pth --input_list /home/face-spoofing/ChaLearn_liveness_challenge/data/lists/folds_by_fakes/fold1/val_list.txt --output_list /home/face-spoofing/ChaLearn_liveness_challenge/output.txt

Cheers!

**Comment by harisudhan312** [comment]
> 
> 
> Hi @John1231983
> I also successfully run the code. Probably this note can be useful:
> 
>     1. Remark all mobilenetv2 import (there are three files, find using "find in file" option on visual code)
> 
>     2. Make sure all paths are correct, including "data_root" at init_dataloader.py, weights paths at resnet_caffe_*.py, and etc.
> 
>     3. Remark this part on "trainer.py":
>        """ self.visdom_log_file = os.path.join(self.opt.out_path, 'log_files', 'visdom.log')
>        self.vis = Visdom(port = opt.visdom_port,
>        log_to_filename=self.visdom_log_file,
>        env=opt.exp_name + '_' + str(opt.fold))
>        ```
>         self.vis_loss_opts = {'xlabel': 'epoch', 
>                               'ylabel': 'loss', 
>                               'title':'losses', 
>                               'legend': ['train_loss', 'val_loss']}
>        
>         self.vis_tpr_opts = {'xlabel': 'epoch', 
>                               'ylabel': 'tpr', 
>                               'title':'val_tpr', 
>                               'legend': ['tpr@fpr10-2', 'tpr@fpr10-3', 'tpr@fpr10-4']}
>        
>         self.vis_epochloss_opts = {'xlabel': 'epoch', 
>                               'ylabel': 'loss', 
>                               'title':'epoch_losses', 
>                               'legend': ['train_loss', 'val_loss']} """
>        ```
> 
>     4. Don't forget to download the model*.pth from CASIA-SURF dataset, and put on the corresponding folder as shown on point 5 (pth path).
> 
>     5. I run the val script as follows:
> 
> 
> python inference.py --config /home/face-spoofing/ChaLearn_liveness_challenge/data/opts/exp1_2stage_seed1/exp1_2stage_seed1_fold1_stage1.opt --pth /home/face-spoofing/ChaLearn_liveness_challenge/data/opts/exp1_2stage_seed1/fold1/checkpoints/model_30.pth --input_list /home/face-spoofing/ChaLearn_liveness_challenge/data/lists/folds_by_fakes/fold1/val_list.txt --output_list /home/face-spoofing/ChaLearn_liveness_challenge/output.txt
> 
> Cheers!

Hi,

I found the links for exp1_2stage and exp3c on google drive to be broken. Can the author check and fix the broken links?

**Comment by harisudhan312** [comment]
@AlexanderParkin 

Hi,

I found the links for exp1_2stage and exp3c on google drive to be broken. Can the author check and fix the broken links?

**Comment by zhangxiaopang88** [comment]
Hello, download the model in the fourth article you wrote. Where did you download the model? could you tell me the specific path? @mrkstt

**Comment by cakuba** [comment]
Hi, could you please kindly tell me where I can find "the model*.pth from CASIA-SURF dataset" as you mentioned? @mrkstt I cannot locate it anywhere within CASIA-SURF dataset, or I am missing something here...

**Comment by mrkstt** [comment]
I think the authors removed the .pth model file. If you're interested, please drop an email to me kahlil@ieee.org @cakuba @zhangxiaopang88 

Thanks and sorry for the late reply.
